astroquery/xmm_newton/core.py

```diff
@@ -0,0 +1,343 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+
+@author: Elena Colomo
+@contact: ecolomo@esa.int
+
+European Space Astronomy Centre (ESAC)
+European Space Agency (ESA)
+
+Created on 3 Sept 2019
+
+
+"""
+from astroquery.utils import commons
+from astropy import units
+from astropy.units import Quantity
+from astroquery.utils.tap.core import TapPlus
+from astroquery.utils.tap.model import modelutils
+from astroquery.query import BaseQuery
+from astropy.table import Table
+from six import BytesIO
+import os
+import re
+
+from . import conf
+from astropy import log
+
+
+__all__ = ['XMMNewton', 'XMMNewtonClass']
+
+
+class XMMNewtonHandler(BaseQuery):
+
+    def __init__(self):
+        super(XMMNewtonHandler, self).__init__()
+
+    def get_file(self, filename, response, verbose=False):
```

Comment:
This function doesn't match the astroquery API, and it's just a very light wrapper that dumps a `response.content` to disk.  Same with `get_table` below.  

Examples like [`retrieve_data`](https://github.com/astropy/astroquery/blob/master/astroquery/eso/core.py#L632) or [`download_files`](https://github.com/astropy/astroquery/blob/master/astroquery/alma/core.py#L403) can be followed, but `get_<x>` implies something different from "download file to disk".

Reply:
Note also that there is a [`_download_file`](https://github.com/astropy/astroquery/blob/master/astroquery/query.py#L251) function in the base `query.py` that handles partial downloads, etc, and should be preferred over direct retrieval of `response.content`